glean-core/rlb/src/test.rs

```diff
@@ -102,12 +160,16 @@ fn initialize_must_not_crash_if_data_dir_is_messed_up() {
         max_events: None,
         delay_ping_lifetime_io: false,
         channel: Some("testing".into()),
+        server_endpoint: Some("invalid-test-host".into()),
+        uploader: None,
     };
 
-    initialize(cfg, ClientInfoMetrics::unknown());
-
-    // TODO: is this test working? is it waiting for init to finish?
-    dispatcher::block_on_queue();
+    reset_glean(cfg, ClientInfoMetrics::unknown(), false);
+    // Glean init is async and, for this test, it bails out early due to
+    // an error: we can do nothing but wait. Tests in other bindings use
```

Comment:
Hm, which error? Shouldn't it still run initialize to completion, thus unblocking the queue, which means we could wait for that?

Reply:
Nope, this test checks that we don't crash if the data directory can't be created. If we can't do that, the glean object instantiation is not created and fails [here](https://searchfox.org/glean/rev/3999e6a5ed45d7b5ed73877c14d4ea4c9855006f/glean-core/rlb/src/lib.rs#142), not ever flushing the queue, as done in the other bindings. let me be more specific in the comment.